next-packages/core/src/keyframes.js

```diff
@@ -1,15 +1,23 @@
 // @flow
-import type { ScopedInsertableStyles } from '@emotion/utils'
+import css from '@emotion/css'
 
 type Keyframes = {|
   name: string,
-  styles: string
-|}
+  styles: string,
+  anim: 1,
+  toString: () => string
+|} & string
 
-export const keyframes = (arg: ScopedInsertableStyles): Keyframes => {
-  const name = `animation-${arg.name}`
+export const keyframes = (...args: *): Keyframes => {
+  let insertable = css(...args)
```

Comment:
could u explain how this works now and how did it work before (im curious about the reasons behind this)?

Reply:
im interested in the design reasons why emotion@9 has returned a class name (string) from `css` call, but emotion@10 is switching to outputting objects, isnt it bad-ish for GC? I realize that GC could be pressured previously in other places

Reply:
By this, i'm assuming you mean keyframes?

Previously, they were used like this
```jsx
/** @jsx jsx */
import { keyframes, css, jsx } from "@emotion/core";

let animation = keyframes(css(...someKeyframes));

render(
  <div
    css={css`
      animtion: ${animation.name};
      ${animation.styles};
    `}
  />
);
```
but this is inconvenient since you have to interpolate the name and the styles and you can't define keyframes inline in a css call.

So this is how it works in this PR.
```jsx
/** @jsx jsx */
import { keyframes, css, jsx } from "@emotion/core";

let animation = keyframes(...someKeyframes);

render(
  <div
    css={css`
      animtion: ${animation};
    `}
  />
);
```

emotion@10 is returning objects from css because it only inserts styles inside the react tree.

While it will likely be marginally worse for the GC, it'll be better for memory usage because of the optimizations in #767 and part of those optimizations relies on information that's only available inside the React tree and the fact that we have objects to get the hash and styles from rather than storing large strings on a cache which isn't great in terms of memory usage.